.github/workflows/cmake.yml

```diff
@@ -0,0 +1,91 @@
+# inspired by "build GDAL library and to run test with CMake"
+# from https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml
+
+name: Build and test with CMake
+
+on:
+    push:
+        paths-ignore:
+            - 'doc/**'
+        branches-ignore:
+            - 'backport**'
+    pull_request:
+        paths-ignore:
+            - 'doc/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  CMAKE_UNITY_BUILD: OFF
+
+permissions:
+  contents: read  #  to fetch code (actions/checkout)
+
+jobs:
+
+  build-linux-ubuntu-20.04:
+    runs-on: ubuntu-20.04
+    env:
+      CMAKE_OPTIONS: -DPython_LOOKUP_VERSION=3.8 -DUSE_CCACHE=ON
+      cache-name: cmake-ubuntu-20.04
+    steps:
+    - name: Checkout GRASS
+      uses: actions/checkout@v3
+    - name: Setup cache
+      uses: actions/cache@v3
+      id: cache
+      with:
+          path: ${{ github.workspace }}/.ccache
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ github.base_ref }}${{ github.ref_name }}-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.cache-name }}-${{ github.base_ref }}
+            ${{ runner.os }}-${{ env.cache-name }}
+    - name: Install CMake 3.11.4
+      run: |
+        cd ${GITHUB_WORKSPACE}
+        wget https://github.com/Kitware/CMake/releases/download/v3.11.4/cmake-3.11.4-Linux-x86_64.tar.gz
+        tar xzf cmake-3.11.4-Linux-x86_64.tar.gz
+        echo "CMAKE_DIR=$GITHUB_WORKSPACE/cmake-3.11.4-Linux-x86_64/bin" >> $GITHUB_ENV
+        echo "$GITHUB_WORKSPACE/cmake-3.11.4-Linux-x86_64/bin" >> $GITHUB_PATH
+    - run: |
+        cmake --version
+    - name: Install dependencies
+      run: |
+        sudo add-apt-repository -y ppa:ubuntugis/ubuntugis-unstable
+        sudo apt-get update
+        sudo apt-get install -y -q build-essential \
+         flex make bison gcc libgcc1 g++ ccache \
+         python3 python3-dev \
+         python3-opengl python3-wxgtk4.0 \
+         python3-dateutil libgsl-dev python3-numpy \
+         wx3.0-headers wx-common libwxgtk3.0-gtk3-dev \
+         libwxbase3.0-dev libbz2-dev zlib1g-dev gettext \
+         libtiff5-dev libpnglite-dev libcairo2 libcairo2-dev \
+         sqlite3 libsqlite3-dev \
+         libpq-dev libreadline6-dev libfreetype6-dev \
+         libfftw3-3 libfftw3-dev \
+         libboost-thread-dev libboost-program-options-dev libpdal-dev\
+         subversion libzstd-dev checkinstall \
+         libglu1-mesa-dev libxmu-dev \
+         ghostscript wget -y make ccache g++ 
+        sudo apt-get update -y
+    - name: Configure ccache
+      run: |
+          echo CCACHE_BASEDIR=$PWD >> ${GITHUB_ENV}
+          echo CCACHE_DIR=$PWD/.ccache >> ${GITHUB_ENV}
+          echo CCACHE_MAXSIZE=250M >> ${GITHUB_ENV}
+          ccache -z
+      working-directory: ${{ github.workspace }}
+    - name: Configure
+      run: |
+        mkdir build
```

Comment:
I see in the job logs that removing this caused a failure. It is created automatically, with this syntax, but that syntax only exists since cmake 3.13. So it fails at the call for configuring, but if the mkdir is added back, I expect it to immediately fail for another reason.

https://cmake.org/cmake/help/latest/manual/cmake.1.html#generate-a-project-buildsystem

Reply:
(It is related to the explicit installation of an old cmake version, that I think is not that useful, but only takes 1second in CI to run)

Reply:
Was it not working before this commit then?

Reply:
With that exact syntax and cmake version: probably. I remember suggesting the syntax a year or two ago, maybe in a PR from netelers branch, but I don't remember the part of the specific old version.

I know I read lots and lots for that, really trying to learn CMake (read 200+ page books for it, plus documentation), and was trying on my side (a year or more ago) to create cmakelists using only best practices, and to not rely on existing quirks/legacy bad practices, to see if I would come up with something. But first, I didn't know enough about CMake, nor specific enough about what builds really are, and lastly, didn't know enough of how GRASS works.

Now I understand way more, since I've been building it repeatedly since that time.